source/common/router/rds_impl.cc

```diff
@@ -180,8 +180,21 @@ RouteConfigProviderSharedPtr RdsFactoryImpl::createRdsRouteConfigProvider(
     Server::Configuration::ServerFactoryContext& factory_context, const std::string& stat_prefix,
     Init::Manager& init_manager, ProtoTraitsImpl& proto_traits,
     Rds::RouteConfigProviderManager& manager) {
+  envoy::extensions::filters::network::http_connection_manager::v3::Rds normalized_rds;
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.normalize_rds_provider_config")) {
+    normalized_rds = rds;
+    // Normalization clears RDS providers's parameters which are not important
+    // for actual fetching of routes. When one of those "not important" parameters is changed,
+    // the existing provider should be chosen.
+    // TODO(cpakulski): Consider adding RdsFactoryImpl::normalizeRdsConfigProvider method, if more
+    // fields needs to be cleared.
+    normalized_rds.mutable_config_source()->clear_initial_fetch_timeout();
```

Comment:
I wonder what's the performance penalty of this approach compared to moving out the `initial_fetch_timeout` and moving it back in after the operation.

Not sure if it's worth breaking the constness, but just wanted to think out loud with you.

Reply:
I had the same dilemma. As you noted above, originally I just wanted to save the field, clear it, calculate hash and restore the value, but constness does not allow to do that, so I came up with copy. The thing is that the caller of `RdsFactoryImpl::createRdsRouteConfigProvider` also takes proto as const, so there are at least 2 levels of function calls (maybe even more) which needs to be changed to eliminate const attribute.

As per performance, in my opinion, the proto message is rather small, as it contains only where to take the routes from, not the routes itself.

I am sure that if I write the microbenchmark, results will point to some degradation of performance, but I also believe that rds updates are not happening all that often.

Reply:
Reverted this change to original code.